### DISCUSSION TITLE
After a 4 day index period, and 20GB of cache/sidecar/config being built... There are no photos found!

### DISCUSSION THREAD

**Discussion by flashvenom** [photoprism/photoprism#3853] [Q&A]
During the 4-day index, I would periodically click on the videos tabs, places, etc and saw that things were percolating.  I never saw the index actually finish but on day 4, it was scanning the 2022 folder so I assumed it was all but done since it scanned all my folders in order.

I rebooted a few days ago without even thinking to check if the index finished... But today **I don't see any photos in PhotoPrism.** There's nothing! I double checked my yml files and the index source is still there. And again, I have 20+GB of "storage" files like sidecar cache, etc.

The only thing I can think of is that the index didn't finish, so maybe it got corrupted, or maybe I have too many photos for PhotoPrism? It was also weird that my credentials didn't work just now... I had to get into the docker shell and reset using photoprism passwd - but I doubt that would have messed it up?

I feel defeated but I just hit re-index. The logs ARE showing things like the below repeated a lot which makes me think they still exist in the database and maybe it will go faster (and actually stick this time?). 
```
2023-10-29 20:22:00 time="2023-10-30T03:22:00Z" level=debug msg="index: updated yaml file CIMG0420.yml"
2023-10-29 20:22:00 time="2023-10-30T03:22:00Z" level=info msg="index: uid ps2w0hz2tjlqr0dt restored from CIMG0422.yml"
2023-10-29 20:22:02 time="2023-10-30T03:22:02Z" level=debug msg="media: created no new thumbnails for 1982-2009/CIMG0427.JPG [309.906641ms]"
```

So my BIG question is this:
-What is the preferred method - indexing or importing? Because my home server has a 12GB RAID for my Dropbox (which includes all my photos), but then a 10GB harddrive seperate for redundancy purposes. I was thinking about starting ALL over, and just deleting everything PhotoPrism has done so far, and instead, IMPORTING every photo from my Dropbox, and on the 10GB harddrive, create a new imports folder.

And of course - any ideas what happened here? (I couldn't find any logs files that were old/dated back to the original index a few weeks ago)

Did I do the right thing by hitting re-index? Or should I perhaps TRY something else, like adding separated, additional mount sources - perhaps grouping every 5 years together or something? Or even a mount point for each year? 

Thanks all! As a developer and Home Automation nerd I am so excited to be a part of this community!

**Comment by graciousgrey** [comment]
Hello :wave: 

Common reasons for missing photos are listed in our troubleshooting guide.

https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

If that doesn't help, we need more information about your setup to figure out where the problem might be.
Did you use our docker-compose examples for the installation or another method? What does your volumes section look like? Are all folders located on persistent volumes? What database are you using? Is it possible that PhotoPrism could not access one of the volumes during indexing?

Checking your logs for errors and warnings might give you additional insights.

**Reply by flashvenom** [reply]
I'm used to IIS logs but I can't seem to find a folder with the complete history of logs. Perhaps I need to set retain settings?

Yes I'm using Docker Compose, with a Docker Engine Desktop for Windows installation. I heard Docker for Windows was horrible in the past but has gotten better as of late - I've kinds seem why that would be said.

I use CLI to launch start.bat using thisdocker-compose.yml: https://pastebin.com/NsbuGZXj

That also spins up a mariadb container. Would all the logs be in the mariadb? Is there an easy way to query that db like with a MSSQLMS?

What's odd is that I'm now 24 hours into the re-re-index and the storage/database is actually getting larger again. I would have expected it to stay the same size if it found/used the existing metadata/thumbnails, etc. I'm on 2013 which does seem a bit faster. That's about 23% done with all my photos.

I looked through the troublshoot guide you linked and most of those were for "missing" on "unindexed". In my case, I'm pretty sure it indexed everything (at least 99% of it) but then after a reboot and changing the password... It's like the database was broken / no originals / no index, etc) everything disappeared other than the back-end database/cache data...

Hopefully it doesn't happen again. I'll be keeping a very close eye once it finishes. Do a backup of everything if it's working before attempting to play/organize/reboot/etc.

My home server is getting pretty slammed. I'm running a Plex server, Sonarr, Radarr, Dropbox, FTP Server, and SMB share and now docker engine with PhotoPrism. All with i3-4360 and 8GB RAM. I probably need to quadruple that ram amount at the very least. One day I'll probably set up a decucated NAS. And then a decicated docker/hyper-v. I'll stop my ramble now before I go on to explain my raspberry pis, linux box, etc etc. Thank you!

**Reply by graciousgrey** [reply]
You can watch the logs as described [here](https://docs.photoprism.app/getting-started/troubleshooting/logs/#__tabbed_1_3). I would recommend to turn on debug logs and observe the logs during indexing.

One thing I noticed in your docker-compose file. It seems like your import folder is configured to be inside your originals folder, which should not be the case. See also:
https://docs.photoprism.app/getting-started/docker-compose/#volumes

Also make sure that your configured storage folder "./storage" is a persistent volume. Otherwise everything that is saved there might be lost when stopping the container.
